Add a brief overview of Dart DevTools to dart run documentation (#4335)
dart-lang/site-www: src/tools/dart-run.md

## Debugging

> To enable debugging, pass one or more debugging options.

Reviewer: **suggestion:** I would show all of the arguments together to help illustrate possibilities, then explain what each does.

```terminal
$ dart run \
   [--enable-asserts FILEPATH] \
   [--observe] \
   [--pause-isolates-on-start]
```

This is consistent with the [Google Style Guide](https://developers.google.com/style/code-syntax#optional-arguments).

Author: If we want to go this direction, it seems more like a complete reference (rather than a small debugging tutorial) and we would maybe need to identify each debug flag, rather than just a few. Is that something we should do for this section?

Reviewer: **comment:** This section does call out three specific flags. It would help to see them together in context. That said, a slight change the first sentence can limit your scope.
```suggestion
To enable debugging, add one or more of these common debugging options to your `dart run` command:
```

> 
> To enable [`assert` statements][assert],
> specify the `--enable-asserts` flag:
> 
> ```terminal
> $ dart run --enable-asserts tool/debug.dart
> ```
> 
> To enable debugging through [Dart DevTools](/tools/dart-devtools),
> a set of debugging and performance tools,
> specify the `--observe` flag.

Reviewer: **issue:** These are a series of non-exclusive options. Consider a list here.
```suggestion
- To enable [`assert` statements][assert],
add the `--enable-asserts` flag.

  \```terminal
  $ dart run --enable-asserts tool/debug.dart
  \```

- To enable debugging through [Dart DevTools](/tools/dart-devtools)
  add the `--observe` flag.
```

Author: I was trying to structure this section more as a mini-tutorial rather than as a reference as we don't include all the flags here, particularly since it usually makes sense to use `--observe` and `--pause-isolates-on-start` together. If we want to adjust to this more reference style, how should we keep that association without requiring users to refer to multiple different parts.

Reviewer: **comment:** If you don't want to provide all debugging flags here (and there's no reason to do so), then modify the intro paragraph. I'll make a note there.

You can also add a graf at the end that says:
```
To learn more about debugging options, see [Dart debugging](link).
```

> Optionally add `--pause-isolates-on-start`
> to break execution at the start of the specified script:

Reviewer: **issue:** For compound sentences, lead with the _why_, then the _how_. Also, you don't need _optionally_. All of these commands are optional.

```suggestion
To break execution at the start of the specified script, add `--pause-isolates-on-start`.
```

```terminal
$ dart run --pause-isolates-on-start --observe tool/debug.dart
```

> 

Reviewer: **issue:** The resulting link is not shown here. It would help understanding to include it.

```terminal
The Dart VM service is listening on http://127.0.0.1:8181/afZySiNbDPg=/
/?uri=ws%3A%2F%2F127.0.0.1%3A8181%2FafZySiNbDPg
```

> 

Reviewer: **issue:** Add a line about what the command returns:
```suggestion
The `dart run` command returns a response that resembles the following:
```

> Open Devtools

Reviewer 2: The _"Open DevTools with the printed **Dart DevTools debugger and profiler** link."_-phrasing is a bit fuzzy.

Can we say something like "Open DevTools by copying the Dart DevTools debugger and profiler URL printed in the terminal, and pasting it into a Chrome browser window"> ?

Reviewer: **issue:** For clarity, I would replace this with:
```
Copy the URL in the response and paste it into a Chrome browser window. Chrome then displays the Dart DevTools Debugger and Profiler.
```

with the printed **Dart DevTools debugger and profiler** link.

> To learn more about using Devtools to debug and profile your app,

Reviewer 2: `DevTools` not `Devtools`

see the rest of the [DartDevTools documentation](/tools/dart-devtools).

> See `dart run --help` for details on other debug options.

Reviewer: **issue:** `See` is somewhat passive and not a call to action. Try:
```suggestion
To learn more about other debug options, run `dart run --help`.
```
